### PR TITLE
Multiple ip preventions

### PR DESCRIPTION
**Pull request by charliekelly13**

### CODE REVIEW

File: src/main/java/universityWebApp/service/LoginAttemptService.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+public class LoginAttemptService {
+
+    @Autowired
+    BlacklistRepository blacklistRepository;
+
+    @Autowired
+    private HttpServletRequest request;
+
+    public LoginAttemptService() {
+        super();
```
Comment (oisinq) [inline src/main/java/universityWebApp/service/LoginAttemptService.java:24]: Do you need a constructor? You've no inheritance, so super() does nothing.

File: src/main/java/universityWebApp/service/LoginAttemptService.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    public void loginSucceeded(String key) {
+        Blacklist black;
+        try {
+            black = blacklistRepository.findById(getIP()).orElseThrow(IpNotFoundException::new);
```
Comment (oisinq) [inline src/main/java/universityWebApp/service/LoginAttemptService.java:30]: nit: do we need to use exceptions here? exceptions == errors to me. could we use optionals instead? not a big deal either way

File: src/main/java/universityWebApp/service/LoginAttemptService.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            black = new Blacklist(getIP(), 1);
+            blacklistRepository.save(black);
+        }
+
+    }
+
+    public boolean isBlocked(String key) {
+        try {
+            Blacklist black =  blacklistRepository.findById(getIP()).orElseThrow(IpNotFoundException::new);
+            return black.getAttempts() >= 3;
```
Comment (oisinq) [inline src/main/java/universityWebApp/service/LoginAttemptService.java:58]: would be nice for this to be a constant at the top of the class

File: src/main/java/universityWebApp/service/LoginAttemptService.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        catch(IpNotFoundException i) {
+            black = new Blacklist(getIP(), 1);
+            blacklistRepository.save(black);
+        }
+
+    }
+
+    public boolean isBlocked(String key) {
+        try {
+            Blacklist black =  blacklistRepository.findById(getIP()).orElseThrow(IpNotFoundException::new);
```
Comment (oisinq) [inline src/main/java/universityWebApp/service/LoginAttemptService.java:57]: ...yeah it makes sense to use an exception here so idk

File: src/main/java/universityWebApp/service/LoginAttemptService.java (hunk 2 above)
Comment (charliekelly13) [reply src/main/java/universityWebApp/service/LoginAttemptService.java:30]: I don't understand what you are saying here

**Comment by oisinq** [comment]
closes #51

**Pull request merged**